### PR TITLE
Bug 1815637 – RTL support 

### PR DESCRIPTION
**Pull request by mavduevskiy**
The original [patch](https://github.com/mozilla-mobile/firefox-android/pull/825) was causing a [crash](https://bugzilla.mozilla.org/show_bug.cgi?id=1839239) on older devices.

Basically, on older versions (<24) the request feature [call](https://github.com/AndroidSDKSources/android-sdk-sources-for-api-level-22/blob/master/com/android/internal/app/AlertController.java#L233) took place during setting the view contents. The extension function for button positioning calls `findViewByID`, which access the decor view and initializes the view partially even before the content was added by the controller. 

Starting from API 25 the call was moved into the [constructor](https://github.com/AndroidSDKSources/android-sdk-sources-for-api-level-24/blob/master/com/android/internal/app/AlertController.java#L209).

The fix was to call the button positioning function after the `DialogFragment` has called `mDialog.show();` and the `AlertController` had time to initialize the view on it's own terms.

### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1815637

### CODE REVIEW

File: fenix/app/src/main/java/org/mozilla/fenix/share/AddNewDeviceFragment.kt [hunk 1]
```diff
@@ -44,7 +45,7 @@ class AddNewDeviceFragment : Fragment(R.layout.fragment_add_new_device) {
             AlertDialog.Builder(requireContext()).apply {
                 setMessage(R.string.sync_connect_device_dialog)
                 setPositiveButton(R.string.sync_confirmation_button) { dialog, _ -> dialog.cancel() }
-                create()
+                create().withCenterAlignedButtons()
```
Comment (Amejia481) [inline fenix/app/src/main/java/org/mozilla/fenix/share/AddNewDeviceFragment.kt:48]: I would like to confirm if this call it's safe? As it looks like we are calling `withCenterAlignedButtons()` before `show()`

File: fenix/app/src/main/java/org/mozilla/fenix/settings/studies/StudiesView.kt [hunk 2]
```diff
@@ -82,7 +83,7 @@ class StudiesView(
                 .setTitle(R.string.preference_experiments_2)
                 .setMessage(R.string.studies_restart_app)
                 .setCancelable(false)
-            val alertDialog: AlertDialog = builder.create()
+            val alertDialog: AlertDialog = builder.create().withCenterAlignedButtons()
             alertDialog.show()
```
Comment (Amejia481) [inline fenix/app/src/main/java/org/mozilla/fenix/settings/studies/StudiesView.kt:87]: Same as https://github.com/mozilla-mobile/firefox-android/pull/3016/files#r1281199075

File: fenix/app/src/main/java/org/mozilla/fenix/settings/studies/StudiesAdapter.kt [hunk 3]
```diff
@@ -141,7 +142,7 @@ class StudiesAdapter(
             .setTitle(R.string.preference_experiments_2)
             .setMessage(R.string.studies_restart_app)
             .setCancelable(false)
-        val alertDialog: AlertDialog = builder.create()
+        val alertDialog: AlertDialog = builder.create().withCenterAlignedButtons()
         alertDialog.show()
```
Comment (Amejia481) [inline fenix/app/src/main/java/org/mozilla/fenix/settings/studies/StudiesAdapter.kt:146]: Same as https://github.com/mozilla-mobile/firefox-android/pull/3016/files#r1281199075

File: fenix/app/src/main/java/org/mozilla/fenix/settings/quicksettings/ClearSiteDataView.kt [hunk 4]
```diff
@@ -107,7 +108,7 @@ class ClearSiteDataView(
                 it.dismiss()
                 interactor.onClearSiteDataClicked(baseDomain)
             }
-            create()
+            create().withCenterAlignedButtons()
```
Comment (Amejia481) [inline fenix/app/src/main/java/org/mozilla/fenix/settings/quicksettings/ClearSiteDataView.kt:111]: Same as https://github.com/mozilla-mobile/firefox-android/pull/3016/files#r1281199075

File: fenix/app/src/main/java/org/mozilla/fenix/settings/logins/fragment/SavedLoginsAuthFragment.kt [hunk 5]
```diff
@@ -215,7 +216,7 @@ class SavedLoginsAuthFragment : PreferenceFragmentCompat() {
                 val intent = Intent(ACTION_SECURITY_SETTINGS)
                 startActivity(intent)
             }
-            create()
+            create().withCenterAlignedButtons()
```
Comment (Amejia481) [inline fenix/app/src/main/java/org/mozilla/fenix/settings/logins/fragment/SavedLoginsAuthFragment.kt:219]: Same as https://github.com/mozilla-mobile/firefox-android/pull/3016/files#r1281199075

**Comment by mavduevskiy** [comment]
@Amejia481 
That's a good question! Yes, it's safe. Only `TimePickerDialogFragment` is using the system `android.app.AlertDialog` while others utilize the `appcompat` version, that has the request feature call in the constructor as well ([link](https://androidx.tech/artifacts/appcompat/appcompat/1.0.2-source/androidx/appcompat/app/AlertController.java.html) for convenience)

**Comment by Amejia481** [comment]
Thank you for the clarification!
In general it looks good, I will run it locally just for a sanity check :)

File: android-components/components/feature/prompts/src/main/java/mozilla/components/feature/prompts/dialog/TimePickerDialogFragment.kt [hunk 6]
```diff
@@ -125,6 +126,17 @@ internal class TimePickerDialogFragment :
...
     }
 
+    override fun onStart() {
+        super.onStart()
+
+        val alertDialog = dialog
+        if (alertDialog is AlertDialog) {
+            // We want to call the extension function after the show() call on the dialog,
+            // and the DialogFragment does that call during onStart().
+            alertDialog.withCenterAlignedButtons()
```
Comment (Amejia481) [inline android-components/components/feature/prompts/src/main/java/mozilla/components/feature/prompts/dialog/TimePickerDialogFragment.kt:136]: Maybe we should add this warning on of the extension documentation as a warning for future consumers.

File: android-components/components/ui/widgets/src/main/java/mozilla/components/ui/widgets/Extentions.kt [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+    findViewById<TextView>(android.R.id.button1)?.let { it.textAlignment = CENTER }
+    findViewById<TextView>(android.R.id.button2)?.let { it.textAlignment = CENTER }
+    findViewById<TextView>(android.R.id.button3)?.let { it.textAlignment = CENTER }
+    return this
+}
+
+/**
+ * A shortcut to align buttons' text to center inside AlertDialog.
+ */
+fun android.app.AlertDialog.withCenterAlignedButtons(): android.app.AlertDialog {
```
Comment (Amejia481) [inline android-components/components/ui/widgets/src/main/java/mozilla/components/ui/widgets/Extentions.kt:23]: Maybe here we should add a warning for future consumers

**Comment by mavduevskiy** [comment]
@Mergifyio backport releases_v117

**Pull request merged**